### PR TITLE
feat: deeptools plotcoverage

### PR DESCRIPTION
**Pull request by tdayris**
### Description

Add Deeptools plotcoverage to the list of available wrappers

### QC

* [X] I confirm that:

For all wrappers added by this PR, 

* there is a test case which covers any introduced changes,
* `input:` and `output:` file paths in the resulting rule can be changed arbitrarily,
* either the wrapper can only use a single core, or the example rule contains a `threads: x` statement with `x` being a reasonable default,
* rule names in the test case are in [snake_case](https://en.wikipedia.org/wiki/Snake_case) and somehow tell what the rule is about or match the tools purpose or name (e.g., `map_reads` for a step that maps reads),
* all `environment.yaml` specifications follow [the respective best practices](https://stackoverflow.com/a/64594513/2352071),
* wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`),
* all fields of the example rules in the `Snakefile`s and their entries are explained via comments (`input:`/`output:`/`params:` etc.),
* `stderr` and/or `stdout` are logged correctly (`log:`), depending on the wrapped tool,
* temporary files are either written to a unique hidden folder in the working directory, or (better) stored where the Python function `tempfile.gettempdir()` points to (see [here](https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir); this also means that using any Python `tempfile` default behavior works),
* the `meta.yaml` contains a link to the documentation of the respective tool or command,
* `Snakefile`s pass the linting (`snakemake --lint`),
* `Snakefile`s are formatted with [snakefmt](https://github.com/snakemake/snakefmt),
* Python wrapper scripts are formatted with [black](https://black.readthedocs.io).
* Conda environments use a minimal amount of channels, in recommended ordering. E.g. for bioconda, use (conda-forge, bioconda, nodefaults, as conda-forge should have highest priority and defaults channels are usually not needed because most packages are in conda-forge nowadays).

### CODE REVIEW

**Comment by fgvieira** [comment]
It would be nice if the wrapper could infer the output plot file format and set `--plotFileFormat`.

**Comment by tdayris** [comment]
Hi, thank you for your review.

The file format is automatically inferred from output file with deepTools. The [documentation](https://deeptools.readthedocs.io/en/develop/content/tools/plotProfile.html#Required%20arguments) reads:

```
--outFileName, -out, -o
  | File name to save the image to. The file ending will  be used to determine the image format. The available options are: “png”,  “eps”, “pdf” and “svg”, e.g., MyHeatmap.png.
```

Let me make the changes anyway. This will avoid all ambiguity and will let the wrapper raise the format error (if any) instead of deepTools.

File: bio/deeptools/plotcoverage/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: deepTools plotCoverage
+url: https://deeptools.readthedocs.io/en/develop/content/tools/plotCoverage.html
+description: >
+  ``deepTools plotCoverage`` assess the sequencing depth of given samples.
+  For more information about ``deepTools``, also see the `source code <https://github.com/deeptools/deepTools>`_.
+authors:
+  - Thibault Dayris
+input:
+  - bam: Path to alignment (BAM)
```
Comment (fgvieira) [inline bio/deeptools/plotcoverage/meta.yaml:9]:
```suggestion
  - bams: Path to alignment (BAM)
```

File: bio/deeptools/plotcoverage/wrapper.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+metrics = snakemake.output.get("metrics", "")
+if metrics:
+    metrics = " --outCoverageMetrics " + metrics
+    if not "--coverageThresholds" in extra:
+        raise ValueError(
+            "Coverage metrics without a `--coverageThresholds` in "
+            "extra parameters will result in an empty file. Please "
+            "provide `--coverageThresholds` or remove "
+            "metrics file from expected output files."
+        )
```
Comment (fgvieira) [inline bio/deeptools/plotcoverage/wrapper.py:31]: If `--coverageThresholds` is needed with `--outCoverageMetrics`, why not just add it also to the command line?

**Review by johanneskoester — APPROVED** [review]

**Pull request merged**